tree/20_cfe_basics/ncf_lib.cf

```diff
@@ -879,14 +879,14 @@ body classes classes_generic_return_boolean_list_two(x, y, boolean_prefix, true_
 # Define x and y prefixed/suffixed with promise outcome
 body classes classes_generic_two(x,y)
 {
-  !has_promiser_stack::
+#  !has_promiser_stack::
     promise_repaired => { "promise_repaired_$(x)", "$(x)_repaired", "$(x)_ok", "$(x)_reached" };
     repair_failed => { "repair_failed_$(x)", "$(x)_failed", "$(x)_not_ok", "$(x)_error", "$(x)_not_kept", "$(x)_not_repaired", "$(x)_reached" };
     repair_denied => { "repair_denied_$(x)", "$(x)_denied", "$(x)_not_ok", "$(x)_error", "$(x)_not_kept", "$(x)_not_repaired", "$(x)_reached" };
     repair_timeout => { "repair_timeout_$(x)", "$(x)_timeout", "$(x)_not_ok", "$(x)_error", "$(x)_not_kept", "$(x)_not_repaired", "$(x)_reached" };
     promise_kept => { "promise_kept_$(x)", "$(x)_kept", "$(x)_ok", "$(x)_not_repaired", "$(x)_reached" };
 
-  has_promiser_stack::
+#  has_promiser_stack::
```

Comment:
we should drop has_promiser_stack, and maybe define a class in the generic method that advertise that it use the class_prefix